index.html

```diff
@@ -83,95 +69,172 @@
+      gx = this.svg.append("g")
+      .attr("class", "x axis")
+      .call(xAxis);
+      
+    gx.select('g').filter(function(d) { return d; })
+        .classed("minor", true);
+    gy.selectAll("text")
+    .attr("x", 4)
+    .attr("dy", -4);
+      // EVENT HANDLERS
+    d3.select('#location').on('change', function () {
+      options.location = d3.event.target.value;
+
+
+      chart.tooltip = d3.tip()
+         .attr('class', 'tooltip')
+         .offset([-6, 0])
+         .html(function(d) { return d.length });
+ 
+     chart.tooltip(chart.svg);
+ 
+     chart.update();
     });
   }
-
-  function update() {
+    Chart.prototype.update = function() {
+      var chart = this;
     // TRANSFORM
     txData = data.slice();
-
-    if (options.yearRange !== 'all') {
-      var yearRange = options.yearRange.split('-');
-      txData = txData.filter(function (d) {
-        return d.year >= yearRange[0] && d.year <= yearRange[1];
+    if (options.location !== 'all') {
+       txData = txData.filter(function (d) {
+        return d.location === options.location;
       });
     }
-
     // SCATTERPLOT
-
-    var points = svg.selectAll('.point')
+    points = this.svg.selectAll('.point')
       .data(txData);
-
     points.enter().append('circle')
       .attr('class', 'point')
       .attr('r', 5);
-
     points
-      .attr('cx', function (d) { return x(d.unemployment); })
-      .attr('cy', function (d) { return y(d.inflation); });
-
+      .attr('cx', function (d) { return x(d.year); })
+      .attr('cy', function (d) { return y(d.value); });
     points.exit().remove();
-
     // HISTOGRAMS
-
+       hist = d3.layout.histogram()
+       .value(function (d) { return d.year; })
+       .bins(chart.x.ticks(binCount))
+       (txData);
+        histy = d3.layout.histogram()
+        .value(function (d) { return d.value; })
+        .bins(chart.y.ticks(binCounty))
+        hy = this.d3.scale.linear()
+        .domain([0, d3.max(hist, function (d) { return d.length; })])
+        .range([0, height]);
+        barWidth = chart.x(hist[1].x) - chart.x(hist[0].x) + 1;
+        bars = chart.svg.selectAll('bar')
+        .data(hist);
+    bars.enter().append('rect')
+      .attr('class', 'bar')
+      .attr('x', function (d) { return x(d.x); })
+      .attr('width', barWidth);
+      .on('mouseover', chart.tooltip.show)
+      .on('mouseout', chart.tooltip.hide);;
+    
+    bars 
+      .attr('y', function (d) { return height - hy(d.length)/4; })
+      .attr('height', function (d) { return hy(d.length)/4; });
+    var barsy = chart.svg.selectAll('bar')
+        .data(hist);
+    bars.enter().append('rect')
+      .attr('class', 'bar')
+      .attr('y', function (d) { return y(d.y); })
+      .attr('width', barWidth);
+      .on('mouseover', chart.tooltip.show)
+      .on('mouseout', chart.tooltip.hide);
+    
+    bars 
+      .attr('y', function (d) { return height - hy(d.length); })
+      .attr('height', function (d) { return hy(d.length); });
```

Comment:
Looks like you started on the `barsy`, but note that this is the same as the code above. Fortunately, as it turns out, as the data is still `hist`.

Remember to move incrementally, checking your browser after each change. If you add a whole bunch of code and something breaks, troubleshooting becomes more difficult because any number of things could have caused it.

It's also a particularly risky endeavor to attempt to refactor non-working code. When we use the word "refactor", we mean to take working code and change its structure so that we can better extend it in the future. If our code never worked, though, we're not yet ready to refactor it, or else we risk burying bugs inside bugs.